packages/searchkit/src/__test__/core/SearchkitManagerSpec.ts

```diff
@@ -1,495 +1,446 @@
-    spyOn(searchkit.accessors, "setState")
-    spyOn(searchkit, "_search")
+    expect(SearchkitManager.prototype.unlistenHistory).toHaveBeenCalled()
+    spyOn(searchkit.accessors, 'setState')
+    spyOn(searchkit, '_search')
     searchkit.completeRegistration()
-    setTimeout(()=> {
+    setTimeout(() => {
       expect(searchkit._search).toHaveBeenCalled()
-      expect(searchkit.accessors.setState)
-        .toHaveBeenCalledWith({q:"foo"})
+      expect(searchkit.accessors.setState).toHaveBeenCalledWith({ q: 'foo' })
       searchkit.unlistenHistory()
       done()
     }, 0)
   })
 
-  it("listenToHistory() - searchOnLoad false", (done)=> {
+  it('listenToHistory() - searchOnLoad false', (done) => {
     const history = createHistoryInstance()
-    history.push(window.location.pathname + "?q=foo-previous")
-    history.push(window.location.pathname + "?q=foo-now")
-    setTimeout(()=> {      
-      const searchkit = new SearchkitManager("/", {
-        useHistory:true,
-        searchOnLoad:false
+    history.push(window.location.pathname + '?q=foo-previous')
+    history.push(window.location.pathname + '?q=foo-now')
+    setTimeout(() => {
+      const searchkit = new SearchkitManager('/', {
+        useHistory: true,
+        searchOnLoad: false
       })
-      spyOn(searchkit.accessors, "setState")
-      spyOn(searchkit, "searchFromUrlQuery")
-      spyOn(searchkit, "_search")
+      spyOn(searchkit.accessors, 'setState')
+      spyOn(searchkit, 'searchFromUrlQuery')
+      spyOn(searchkit, '_search')
       searchkit.setupListeners()
-      searchkit.completeRegistration()    
-      setTimeout(()=> {
+      searchkit.completeRegistration()
+      setTimeout(() => {
         history.goBack()
-        setTimeout(()=> {
-          expect(searchkit.searchFromUrlQuery).toHaveBeenCalledWith("?q=foo-previous")
+        setTimeout(() => {
+          expect(searchkit.searchFromUrlQuery).toHaveBeenCalledWith('?q=foo-previous')
           searchkit.unlistenHistory()
           done()
-        },10)      
-        },0)
+        }, 10)
       }, 0)
-
+    }, 0)
   })
 
-  it("listenToHistory() - handle error", (done)=> {
+  it('listenToHistory() - handle error', (done) => {
     const history = createHistoryInstance()
-    history.push(window.location.pathname + "?q=foo")
-    const searchkit = new SearchkitManager("/", {
-      useHistory:true
+    history.push(window.location.pathname + '?q=foo')
+    const searchkit = new SearchkitManager('/', {
+      useHistory: true
     })
     searchkit.setupListeners()
 
-    const error = new Error("oh no")
+    const error = new Error('oh no')
     searchkit.searchFromUrlQuery = () => {
       throw error
     }
-    spyOn(console, "error")
+    spyOn(console, 'error')
     searchkit.completeRegistration()
-    setTimeout(()=> {
-      const consoleErrorCallArg = console.error["calls"].argsFor(0)[0]
+    setTimeout(() => {
+      const consoleErrorCallArg = console.error['calls'].argsFor(0)[0]
```

Comment:
Is it OK that there is no call to `console.error.restore()`?

Reply:
now using jest spyOn + mockImplementation instead and configured jest to mockRestore after each test